src/flint/nmod.jl

```diff
@@ -56,13 +56,15 @@ is_unit(a::zzModRingElem) = a.parent.n == 1 ? a.data == 0 : gcd(a.data, a.parent
 
 modulus(R::zzModRing) = R.n
 
+characteristic(R::zzModRing) = ZZRingElem(modulus(R))
+
+is_trivial(a::zzModRing) = is_unit(modulus(a))
```

Comment:
Is there a reason you use is_unit here but is_one in the other file?

Reply:
is_unit is mathematically correct. here it gets optimized to just checking for one as the modulus is unsigned . for the ZZ case using isone is an optimization to save 1 or 2 machine instructions checking for -1

Reply:
ah ok. thanks for the explanation